src/components/TextEditor.js

```diff
@@ -0,0 +1,232 @@
+              keyBindingFn={this.mapKeyToEditorCommand}
+              onChange={this.onChange}
+              placeholder="Tell a story..."
+              ref="editor"
+              spellCheck={true}
+            />
+            
+          </div>
+        </div>
+      );
+    }
+  }
+
+  // Custom overrides for "code" style.
+  const styleMap = {
+    CODE: {
+      backgroundColor: 'rgba(0, 0, 0, 0.05)',
+      fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
+      fontSize: 16,
+      padding: 2,
+    },
+  };
+
+  function getBlockStyle(block) {
+    switch (block.getType()) {
+      case 'blockquote': return 'RichEditor-blockquote';
+      default: return null;
+    }
+  }
+
+  class StyleButton extends React.Component {
+    constructor() {
+      super();
+      this.onToggle = (e) => {
+        e.preventDefault();
+        this.props.onToggle(this.props.style);
+      };
+    }
+
+    render() {
+      let className = 'RichEditor-styleButton';
+      if (this.props.active) {
+        className += ' RichEditor-activeButton';
+      }
+
+      return (
+        <span className={className} onMouseDown={this.onToggle}>
+          {this.props.label}
+        </span>
+      );
+    }
+  }
+
+  const BLOCK_TYPES = [
+    {label: 'H1', style: 'header-one'},
+    {label: 'H2', style: 'header-two'},
+    {label: 'H3', style: 'header-three'},
+    {label: 'UL', style: 'unordered-list-item'},
+    {label: 'OL', style: 'ordered-list-item'},
+  ];
+
+  const BlockStyleControls = (props) => {
+    const {editorState} = props;
+    const selection = editorState.getSelection();
+    const blockType = editorState
+      .getCurrentContent()
+      .getBlockForKey(selection.getStartKey())
+      .getType();
+
+    return (
+      <div className="RichEditor-controls">
+        {BLOCK_TYPES.map((type) =>
+          <StyleButton
+            key={type.label}
+            active={type.style === blockType}
+            label={type.label}
+            onToggle={props.onToggle}
+            style={type.style}
+          />
+        )}
+      </div>
+    );
+  };
+
+  var INLINE_STYLES = [
+    {label: 'B', style: 'BOLD', class: 'bold'},
+    {label: 'i', style: 'ITALIC', class: 'italic'},
+    {label: 'U', style: 'UNDERLINE', class: 'underline'},
+  ];
+
+  const InlineStyleControls = (props) => {
+    const currentStyle = props.editorState.getCurrentInlineStyle();
+    
+    return (
+      <div className="RichEditor-controls">
+        {INLINE_STYLES.map((type) =>
+          <div className={type.class}>
+          <StyleButton
+            key={type.label}
+            active={currentStyle.has(type.style)}
+            label={type.label}
+            onToggle={props.onToggle}
+            style={type.style}
+          />
+          </div>
+        )}
+      </div>
+    );
+  };
+
+  function getEditorState(){
+    return DESCRIPTIONRAW;
+  }
+
+  function setEditorState(storedState, hasContent){
```

Comment:
hasContent seems unnecessary. Could the code simply check if there is anything inside storedState argument to decide if there is any content?

Reply:
not really. storedState is still a created editor object. i've tried a few things but it is a lot simpler to do that. there are editor content objects either way. would have to do a lot more complicated checking or just pass the boolean